### DISCUSSION TITLE
copyTo() file size

### DISCUSSION THREAD

**Discussion by magicmatt007** [isar/isar#1430] [Help]
I understood that copyTo() would create a _compacted_ copy of the isar database.

However, if I use copyTo() on a fresh default.isar with only a couple of entries, the size is growing:

default.isar:   1024 kb
backup.isar:   5120 kb

Am I doing something wrong, or is this the expected behaviour?

Thx!

**Comment by abdullah-cse** [comment]
Though it's not related to you question, but you may already know how to create ISAR-Database backup on Phone.
I've tried the following code, but ISAR complains File already exists. But the folder is empty.

```dart
//Get user specified Path
String? path = await FilePicker.platform.getDirectoryPath();

// Create a file
final copiedDbFile = File();

//Now copy Isar Databse to this File
isar.copyToFile(copiedDbFile.path);
```

Is there any proper way to backup in Phone Storage?

**Reply by Jtplouffe** [reply]
You are not passing a path to your newly created file, and you are not using the `path` returned by the file picker.
You should also probably pass a filename to the file, as `getDirectoryPath` probably just returns a path to the directory. You could use the `path` package for that.

It could look something like this:

```dart
import "package:path/path.dart" as path;

//Get user specified Path
String? directoryPath = await FilePicker.platform.getDirectoryPath();

// Create a file
final copiedDbFile = File(path.join(directoryPath, "backup_isar_db"));

//Now copy Isar Databse to this File
isar.copyToFile(copiedDbFile.path);
```

You can refer to the unit tests for more examples: https://github.com/isar/isar/blob/95e1f02c274bb4bb80f98c1a42ddf33f3690a50c/packages/isar_test/test/copy_to_file_test.dart

**Reply by abdullah-cse** [reply]
Thanks for your awesome reply. I solve this by

```dart
//Get user specified Path
String? directoryPath = await FilePicker.platform.getDirectoryPath();

// Create a file in Storage Device named "AppsBackUpData"

final File copiedDbFile = File('$directoryPath/AppsBackUpData');

//Now copy Isar Databse to this File
isar.copyToFile(copiedDbFile.path);
```

**Comment by derrickgw** [comment]
I am seeing the same thing.  However, it doesn't seem to be growing at all.  It has become one way that I know if I am looking at the exported version or the original.
